src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -51,6 +52,31 @@ def str_to_sql(value: str) -> str:
     return f"'{sql_str}'"
 
 
+def str_to_sql_for_year_month_interval(
+    value: str, datatype: YearMonthIntervalType
+) -> str:
+    # Extract interval value from strings like "INTERVAL 1-7 YEAR TO MONTH"
+    # Examples: "1-7" -> INTERVAL '1-7' YEAR TO MONTH, "1" -> INTERVAL '1' YEAR
+    # "1-7" -> INTERVAL '7' MONTH
+    parts = value.split(" ")
+    if len(parts) < 2:
+        raise ValueError(f"Invalid interval format: {value}")
```

Comment:
As a reader, this is still hard to understand. How do we expect value to look like. Should it be `INTERVAL YY-MM [YEAR TO MONTH | YEAR | MONTH]`